### PR TITLE
Update Protocol section in Developer docs [part 3 of X]

### PR DESCRIPTION
**Pull request by espendk**
- add reference to `TickLib` 
- emphasize the literate source code
- update the following pages/sections:
    - "Views on offers"
    - "Taking offers", renamed to "Market order"
    - "Periphery contracts" (first pass, more may come)
- re-org the Protocol -> Technical reference section:
   - After the removal of snipes and introduction of cleanByImpersonation, the old structures around "taker orders" no longer makes sense: The only taker order is a market order.
   - Eg cleaning is not a taker order and offer lists are relevant to more than market orders and offer making.
   - The new structure is a bit too long imo, but more coherent than before.. We can try to group it better later.

### CODE REVIEW

File: docs/developers/protocol/technical-references/market-order/README.md [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+Suppose one wants to buy or sell some token `B` (base), using token `Q` (quote) as payment, on a market with tick spacing `T`.
+
+* **Market buy:** A limit **buy** order for `x` `B` tokens, corresponds to a `marketOrderByTick` on the (`B`,`Q`,`T`) offer list with `fillWants` set to `true`, `fillVolume = x` (the volume one wishes to buy), and `maxTick` = $⌊log_{1.0001}(price\ limit)⌋$.
+* **Market sell:** A limit **sell** order for `x` `B` tokens, corresponds to a `marketOrderByTick` on the (`Q`,`B`,`T`) offer list with
+ `fillWants` set to `false`, `fillVolume = x` (the volume one wishes to sell), and `maxTick` = $⌊-log_{1.0001}(price\ limit)⌋$.
+
+
+:::caution **On order residuals**
+
+Contrary to [GTC orders](https://www.investopedia.com/terms/g/gtc.asp) on regular [order book](https://www.investopedia.com/terms/o/order-book.asp) based exchanges, the residual of your order (i.e., the volume you were not able to buy/sell due to hitting your price limit) will _not_ be put on the market as an offer. Instead, the market order will simply end partially filled.
```
Comment (maxencerb) [inline docs/developers/protocol/technical-references/market-order/README.md:348]: Can't we add a reference there to limit orders for the GTC equivalent there ?

File: docs/developers/strat-lib/guides/HowToTest.md [hunk 2]
```diff
@@ -92,7 +92,7 @@ Let us unfold the specification for the test itself.
 For this first test, we want to 
 
 * post and fund a pair of offers using [`Amplifier.newAmplifiedOffers(...)`](./DirectHowTo.md#publishing-amplified-liquidity), 
-* [snipe](../../protocol/technical-references/taking-and-making-offers/taker-order/README.md#offer-sniping) *one* of the offers, and,
+* snipe *one* of the offers, and,
```
Comment (maxencerb) [inline docs/developers/strat-lib/guides/HowToTest.md:95]: Sniping does not exist anymore

File: docs/developers/strat-lib/guides/HowToTest.md [hunk 3]
```diff
@@ -111,7 +111,7 @@ https://github.com/mangrovedao/mangrove-strats/blob/a265abeb96a053e386d346c7c9e4
 
 #### Taking a Single Offer
 
-Mangrove provides [snipe](../../protocol/technical-references/taking-and-making-offers/taker-order/README.md#offer-sniping) exactly for taking a *specific* offer. 
+Mangrove provides snipe exactly for taking a *specific* offer.
```
Comment (maxencerb) [inline docs/developers/strat-lib/guides/HowToTest.md:114]: Same here as well, no snipe

File: docs/developers/strat-lib/guides/HowToTest.md (hunk 2 above)
Comment (espendk) [reply docs/developers/strat-lib/guides/HowToTest.md:95]: Indeed. However, this PR does not update the strat lib section, only the protocol, so for now I just removed the broken links

File: docs/developers/strat-lib/guides/HowToTest.md (hunk 3 above)
Comment (espendk) [reply docs/developers/strat-lib/guides/HowToTest.md:114]: same answer :-)

File: docs/developers/protocol/technical-references/market-order/README.md (hunk 1 above)
Comment (espendk) [reply docs/developers/protocol/technical-references/market-order/README.md:348]: Good idea. Addressed in c4b7075373158abf24e0fa162825a4133e9690a5

**Review by maxencerb — APPROVED** [review]

**Pull request merged**